src/sage/rings/finite_rings/integer_mod_ring.py

```diff
@@ -1920,6 +1920,34 @@ def _roots_univariate_polynomial(self, f, ring=None, multiplicities=True, algori
             sage: f = R.random_element(degree=5000)
             sage: f.roots(multiplicities=False)
             [107295314027801680550847462044796892009, 75545907600948005385964943744536832524]
+
+        Roots of 0 or 1 degree polynomials should be computable without factoring the order:
+            sage: set_random_seed(31337)
+            sage: p = random_prime(2^512)
+            sage: q = random_prime(2^512)
+            sage: R.<x> = Zmod(p*q)[]
+            sage: R(1).roots(multiplicities=False)
+            []
+            sage: R(x + 1).roots(multiplicities=False)
+            [3370329767268559825901742050737977765362112301264016666995728641595783763682895579491491312949884527233896437825100351818777861111199083287936712213803662588205628332090619063390145822822539640664875016851649375480771418576780687530310519702682544247717838214152013112397973300661385457793537189792695633500]
+            sage: R(p*x + 1).roots(multiplicities=False)
+            []
+
+        Even when leading coefficient of linear polynomial is not invertible:
```

Comment:
```suggestion
        Even when leading coefficient of linear polynomial is not invertible::

```
Same elsewhere. Double colons then new line

Reply:
Double colons don't seem to be required since all other tests use a single colon too. Added new lines though.

Reply:
https://github.com/sagemath/sage/pull/40551 It forms a [literal block](https://docutils.sourceforge.io/docs/ref/doctree.html#literal-block)

Reply:
better to link https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks